### PR TITLE
Implement OpenAI Integration to Slack

### PR DESCRIPTION
**Pull request by arkanalexei**
**Changes Made:**

- Refactored ask method to handle loading message and chatbot response updates more efficiently
- Refactored test setup by introducing reusable fixtures and helper methods to reduce code duplication and improve readability.
- Consolidated Slack API and request mocks for consistent behavior and easier maintenance.

**Testing:**

- Updated and verified unit tests to ensure proper handling of Slack API errors, chatbot responses, and edge cases.

### CODE REVIEW

File: adapter/slack.py [hunk 1]
```diff
@@ -3,12 +3,14 @@
 from slack_bolt import App
 from slack_bolt.adapter.fastapi import SlackRequestHandler
 from slack_sdk.errors import SlackApiError
+from chat.chat import ChatOpenAI
 
 class SlackAdapter:
-    def __init__(self, app: App) -> None:
+    def __init__(self, app: App, chatbot: ChatOpenAI) -> None:
```
Comment (TakeMeGH) [inline adapter/slack.py:9]: So for now, the integration is only with OpenAI, right? If we want to use another chatbot model, do we need to add a new chatbot parameter in the Slack adapter constructor?

File: adapter/slack.py [hunk 2]
```diff
@@ -29,18 +31,26 @@ async def ask(self, request: Request):
         else:
             raise HTTPException(status_code=400, detail="Missing parameter in the request.")
         
-        question = f"{question} \n\n<@{user_id}>"
-        
+        question = f"<@{user_id}> asked: \n\n\"{question}\" "
```
Comment (TakeMeGH) [inline adapter/slack.py:34]: Is there a specific reason for changing the format?

File: adapter/slack.py (hunk 1 above)
Comment (arkanalexei) [reply adapter/slack.py:9]: Great catch, it shouldn't be `ChatOpenAI` but rather a `Chat` class. Since it could be ChatOpenAI or ChatAnthropic, etc. Will fix this soon!

File: adapter/slack.py (hunk 2 above)
Comment (arkanalexei) [reply adapter/slack.py:34]: This is purely subjective but I just thought it looked nicer.

**Review by TakeMeGH — APPROVED** [review]
